api/candlepin-api-spec.yaml

```diff
@@ -8888,8 +8888,8 @@ components:
     EnvironmentContentDTO:
       description: EnvironmentContent represents the promotion of content into a particular environment.
       properties:
-        content:
-          $ref: "#/components/schemas/ContentDTO"
+        contentId:
+          type: string
```

Comment:
Hrmm... investigating the origins of the environment/key cleanup code, it goes back a very long time -- roughly 10 years [1].

So apparently the way refresh worked in the past was what was saving them -- we would unlink products and content from pools as they were cleaned up and deleted, but the products or content themselves would stick around indefinitely. But if the content was ever manually deleted, it would be removed from the environment all the same.

I *really* hate the orphan grace period code and flow, but it seems this change would also introduce weird behavior relative to past versions. Going to think on this a bit before fully committing to that cleanup removal; maybe there's a better option. The UUID -> ID change is probably still a net good one, but we'd still need to do some cleanup on the deletion of products and content regardless.

[1] https://github.com/candlepin/candlepin/commit/678f1f0b2f91d1241f693aa3d45b71c391b76ae5